### PR TITLE
Fix XML configuration dictionary containing list of components throws exceptions

### PR DESCRIPTION
**Pull request by ni-mi**
https://github.com/castleproject/Windsor/issues/574
Fix dictionary bug where reference to list components didn't work

### CODE REVIEW

File: src/Castle.Windsor.Tests/Config/ConfigurationTestCase.cs [hunk 1]
```diff
@@ -33,6 +35,98 @@ namespace Castle.MicroKernel.Tests.Configuration
...
+			Container.Install(Configuration.FromXml(new StaticContentResource(config)));
+			var stringsList = Container.Resolve<List<string>>("list");
+			var stringToListDictionary = Container.Resolve<Dictionary<string, List<string>>>("stringToListDictionary");
+			Assert.NotNull(stringToListDictionary);
+			Assert.AreEqual(2, stringToListDictionary.Count);
+		}
+		
+		
+		
+
```
Comment (jonorossi) [inline src/Castle.Windsor.Tests/Config/ConfigurationTestCase.cs:129]: There is a whole bunch of blank lines here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 Bugfixes:
 - Fix CollectionResolver to allow propagation of inline dependencies (@dvdwouwe, #562)
 - Allow DefaultNamingSubSystem derivatives to invalidate the cache which was accidently removed in 5.1.0 (@nativenolde, #569)
+- Fix dictionary bug where reference to list components didn't work (@ni-mi, #575)
```
Comment (jonorossi) [inline CHANGELOG.md:8]: Could you change this to refer to the fact this is about XML configuration, right now it is unclear especially that most people are doing everything in code.

**Comment by jonorossi** [comment]
FYI, your commits are using an email address not linked to your GitHub profile. You might want to check if that is the address you intended to use, and fix if not.

File: src/Castle.Windsor.Tests/Config/ConfigurationTestCase.cs [hunk 3]
```diff
@@ -33,6 +35,98 @@ namespace Castle.MicroKernel.Tests.Configuration
 	[TestFixture]
 	public class ConfigurationTestCase : AbstractContainerTestCase
 	{
+		
+		
+
```
[the remark this answers is not in the document]
Comment (ni-mi) [reply src/Castle.Windsor.Tests/Config/ConfigurationTestCase.cs]: Deleted them. Thanks!

File: src/Castle.Windsor.Tests/Config/ConfigurationTestCase.cs (hunk 1 above)
Comment (ni-mi) [reply src/Castle.Windsor.Tests/Config/ConfigurationTestCase.cs]: Deleted them. Thanks!

File: CHANGELOG.md (hunk 2 above)
Comment (ni-mi) [reply CHANGELOG.md]: Changed the description to reflect the fact that it is about XML configuration.

File: src/Castle.Windsor.Tests/Config/ConfigurationTestCase.cs [hunk 4]
```diff
@@ -33,6 +35,98 @@ namespace Castle.MicroKernel.Tests.Configuration
...
+			Container.Install(Configuration.FromXml(new StaticContentResource(config)));
+			var stringToStringDictionary = Container.Resolve<Dictionary<string, string>>("stringToStringDictionary");
+			Assert.NotNull(stringToStringDictionary);
+			Assert.AreEqual(2, stringToStringDictionary.Count);
+			Assert.AreEqual("Property Value 1", stringToStringDictionary["Key 1"]);
+			Assert.AreEqual("Property Value 2", stringToStringDictionary["Key 2"]);
+		}
+		
+
```
[the remark this answers is not in the document]
Comment (ni-mi) [reply src/Castle.Windsor.Tests/Config/ConfigurationTestCase.cs]: Deleted them. Thanks!

**Comment by ni-mi** [comment]
> FYI, your commits are using an email address not linked to your GitHub profile. You might want to check if that is the address you intended to use, and fix if not.

Thank you, I haven't noticed - changed that.

**Comment by ni-mi** [comment]
@jonorossi  - please re-review :) 
Also, I'd appreciate if you could let me know when you think you can release a version with those changes since they are blockers for us.

Thanks!

**Review by jonorossi — APPROVED** [review]

**Pull request merged**